### PR TITLE
apis: add extension plugin framework for PodMetric

### PR DESCRIPTION
**Pull request by jasonliu747**
Signed-off-by: Jason Liu <jasonliu747@gmail.com>

### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `make test`

### CODE REVIEW

File: apis/slo/v1alpha1/nodemetric_types.go [hunk 1]
```diff
@@ -49,6 +49,8 @@ type PodMetricInfo struct {
 	Name      string      `json:"name,omitempty"`
 	Namespace string      `json:"namespace,omitempty"`
 	PodUsage  ResourceMap `json:"podUsage,omitempty"`
+	// Third party extensions for PodMetric
+	Extensions *MetricExtensions `json:"extensions,omitempty"`
```
Comment (zwzhang0107) [inline apis/slo/v1alpha1/nodemetric_types.go:53]: consider use `ExtensionsMap` in nodeslo_ext.go？

Comment (jasonliu747) [reply apis/slo/v1alpha1/nodemetric_types.go]: SGTM

**Pull request merged**